### PR TITLE
[dy] Move get_repo_path to settings module

### PR DESCRIPTION
**Pull request by dy46**
# Summary

This PR moves a handful of `repo_manager` methods to a `mage_ai.settings.repo` module. These methods are called by a number of files in the repo, but the `repo_manager` module has a lot of dependencies. This creates the possibility of introducing circular dependency errors. I moved the following methods to the settings modules:

* `get_repo_path`
* `set_repo_path`
* `get_repo_name`
* `get_data_dir`

I left the original methods in the `repo_manager` file, but I added a deprecation warning when they are used.

# Tests

tested locally

cc: @wangxiaoyou1993

### CODE REVIEW

File: mage_ai/data_preparation/shared/secrets.py [hunk 1]
```diff
@@ -72,20 +75,20 @@ def get_valid_secrets() -> List:
     return valid_secrets
 
 
-def get_secret_value(name: str, repo_name: str = None) -> str:
+def get_secret_value(name: str, repo_name: str = get_repo_path()) -> str:
```
Comment (wangxiaoyou1993) [inline mage_ai/data_preparation/shared/secrets.py:78]: It'll be better to set the default value to None in method definition. Then inside the method, if repo_name is None, repo_name = get_repo_path().

When the method is defined, repo path might not be set yet.

File: mage_ai/data_preparation/repo_manager.py [hunk 2]
```diff
@@ -248,3 +224,46 @@ def update_project_uuid():
...
+REPO_PATH_ENV_VAR = 'MAGE_REPO_PATH'
+
+
+def get_repo_path() -> str:
+    warn(
+        'repo_manager.get_repo_path is deprecated. Please use mage_ai.settings.repo.get_repo_path',
+        DeprecationWarning,
+        stacklevel=2
+    )
+    return os.getenv(REPO_PATH_ENV_VAR) or os.getcwd()
```
Comment (wangxiaoyou1993) [inline mage_ai/data_preparation/repo_manager.py:241]: You can actually call the method from `mage_ai.settings.repo` here for code consistency.

File: docs/development/testing/unit-tests.mdx [hunk 3]
```diff
@@ -19,7 +19,7 @@ End-to-end test for a pipeline:
 
 ```python
 from mage_ai.data_preparation.models.pipeline import Pipeline
-from mage_ai.tests.base_test import TestCase
+from mage_ai.tests.test_base import TestCase
```
[the remark this answers is not in the document]
Comment (dy46) [reply docs/development/testing/unit-tests.mdx:22]: i was trying to fix some lint issues, but decided against it. i thought i reverted all the changes, but forgot this one

File: mage_ai/data_preparation/repo_manager.py (hunk 2 above)
Comment (dy46) [reply mage_ai/data_preparation/repo_manager.py]: good idea

**Comment by dy46** [comment]
@wangxiaoyou1993 addressed your comments

**Review by wangxiaoyou1993 — APPROVED** [review]

**Pull request merged**